test/powershell/Modules/Microsoft.PowerShell.Core/Where-Object.Tests.ps1

```diff
@@ -142,4 +142,89 @@ Describe "Where-Object" -Tags "CI" {
+            # With -GT:False, the parameter is not specified, so default boolean evaluation applies
+            $result = $testData | Where-Object Age -GT:$false
+            $result | Should -HaveCount 3
+        }
+
+        It "-Like:$false should behave like default boolean evaluation" {
+            # With -Like:False, the parameter is not specified, so default boolean evaluation applies
+            $result = $testData | Where-Object City -Like:$false
+            $result | Should -HaveCount 3
+        }
+
+        It "-Match:$false should behave like default boolean evaluation" {
+            # With -Match:False, the parameter is not specified, so default boolean evaluation applies
+            $result = $testData | Where-Object Name -Match:$false
+            $result | Should -HaveCount 3
+        }
+
+        It "-Contains:$false should behave like default boolean evaluation" {
+            # With -Contains:False, the parameter is not specified, so default boolean evaluation applies
+            # "... | Where-Object Name" checks if Name property is truthy
+            $result = $testData | Where-Object Name -Contains:$false
+            $result | Should -HaveCount 3
+        }
+
+        It "-In:$false should behave like default boolean evaluation" {
+            # With -In:False, the parameter is not specified, so default boolean evaluation applies
+            # "... | Where-Object Name" checks if Name property is truthy
+            $result = $testData | Where-Object Name -In:$false
+            $result | Should -HaveCount 3
+        }
+
+        It "-Is:$false should behave like default boolean evaluation" {
+            # With -Is:False, the parameter is not specified, so default boolean evaluation applies
+            # "... | Where-Object Name" checks if Name property is truthy
+            $result = $testData | Where-Object Name -Is:$false
+            $result | Should -HaveCount 3
+        }
+
+        It "-Not:$false should behave like default boolean evaluation" {
+            $testData2 = @(
+                [PSCustomObject]@{ Name = ''; Value = 1 }
+                [PSCustomObject]@{ Name = 'Test'; Value = 2 }
+            )
+            # With -Not:False, the parameter is not specified, so default boolean evaluation applies
+            # "... | Where-Object Name" returns objects where Name is truthy (non-empty)
+            $result = $testData2 | Where-Object Name -Not:$false
+            $result | Should -HaveCount 1
+            $result.Name | Should -Be 'Test'
+        }
+
+        It "-GT:$false with -Value should throw an error requiring an operator" {
+            # When a switch parameter is set to $false with -Value specified,
+            # it should throw an error because no valid operator is active
+            { $testData | Where-Object Age -GT:$false -Value 25 } | Should -Throw -ErrorId 'OperatorNotSpecified,Microsoft.PowerShell.Commands.WhereObjectCommand'
```

Comment:
> An operator is required to compare the two specified values. Include a valid operator in the command, and then try the command again. For example, Get-Process | Where-Object -Property Name -eq Idle

Hmm, formally it is not correct. @mklement0 Do you expect user's reports about this? It seems we need to update the error message.